datasets/text/text.py

```diff
@@ -11,6 +11,7 @@ def _split_generators(self, dl_manager):
             If str or List[str], then the dataset returns only the 'train' split.
             If dict, then keys should be from the `nlp.Split` enum.
         """
+        assert bool(self.config.data_files), "At least one data file must be specified, but got data_files={}".format(self.config.data_files)
```

Comment:
Let's merge the PR on multi-threaded `text` and add it in the new script before?

Reply:
Be careful by putting all these checks in `assert`s - that's not what they should be used for. ValueErrors or filenotfound errors might be better. the issue with using asserts everywhere is that users will miss out on these error messages when they run their scripts in optimized mode, which turns off asserts (which are basically debug statements). This is true for the whole code base - using asserts to check arguments is not always what you'd want from a usability perspective.

I'd change it to:

```python
if not self.config.data_files:
    raise ValueError(f"At least one data file must be specified, but got data_files={self.config.data_files}")
```

Reply:
Yes indeed